internal/command/meta_backend_migrate.go

```diff
@@ -579,6 +579,20 @@ func (m *Meta) backendMigrateTFC(opts *backendMigrateOpts) error {
 		opts.sourceWorkspace = currentWorkspace
 
 		log.Printf("[INFO] backendMigrateTFC: single-to-single migration from source %s to destination %q", opts.sourceWorkspace, opts.destinationWorkspace)
+
+		// If the current workspace is has no state we do not need to ask
+		// if they want to migrate the state.
+		sourceState, err := opts.Source.StateMgr(currentWorkspace)
+		if err != nil {
+			return err
+		}
+		if err := sourceState.RefreshState(); err != nil {
+			return err
+		}
+		if sourceState.State().Empty() {
+			return nil
```

Comment:
I'd probably add an `[INFO]` log line here to indicate in debug output that the migration (indicated on line 581) was skipped because the workspace was empty.

Reply:
✔️